lib/modules/manager/gitlabci/utils.ts

```diff
@@ -24,14 +24,32 @@ const depProxyRe = regEx(
  * @param imageName as used in .gitlab-ci.yml file
  * @return package dependency for the image
  */
-export function getGitlabDep(imageName: string): PackageDependency {
+export function getGitlabDep(
+  imageName: string,
+  registryAliases?: Record<string, string>
+): PackageDependency {
   const match = depProxyRe.exec(imageName);
   if (match?.groups) {
     const dep = { ...getDep(match.groups.depName), replaceString: imageName };
     // TODO: #7154
     dep.autoReplaceStringTemplate = `${match.groups.prefix}${dep.autoReplaceStringTemplate}`;
     return dep;
-  } else {
-    return getDep(imageName);
   }
+
+  for (const [name, value] of Object.entries(registryAliases ?? {})) {
+    const escapedName = escapeRegExp(name);
+    const match = regEx(`(?<prefix>${escapedName}/)(?<depName>.+)`).exec(
+      imageName
+    );
+    if (match?.groups) {
+      const dep = {
+        ...getDep(`${value}/${match.groups.depName}`),
+        replaceString: imageName,
+      };
+      dep.autoReplaceStringTemplate = `${match.groups.prefix}${dep.autoReplaceStringTemplate}`;
+      return dep;
```

Comment:
i think directly return the dep without variable declaration and previous comment fix.